インストールに関して

東方 Fedora24 を利用して Comainu  のインストールをしようとしています。
/script/setup.sh を実行しているのですが、YamCha のインストールの際に、エラーを吐きます。

```
make  all-recursive
make[1]: ディレクトリ '/home/xxx/yamcha-0.33' に入りmます
Making all in src
make[2]: ディレクトリ '/home/xxx/yamcha-0.33/src' に入ります
/bin/sh ../libtool --mode=compile --tag=CXX g++ -DHAVE_CONFIG_H -I. -I. -I..
-O3 -Wno-deprecated -Wall -c -o feature_index.lo feature_index.cpp
 g++ -DHAVE_CONFIG_H -I. -I. -I.. -O3 -Wno-deprecated -Wall -c feature_index.cpp
-fPIC -DPIC -o .libs/feature_index.o
feature_index.cpp: In member function 'bool
YamCha::FeatureIndex::setFeature(const string&, int)':
feature_index.cpp:150:62: error: no matching function for call to
'make_pair(const int&, const int&)'
    feature_set.insert (std::make_pair <int, int>( *rit, *cit ) );
                                                              ^
In file included from /usr/include/c++/6.2.1/utility:70:0,
                 from /usr/include/c++/6.2.1/tuple:38,
                 from /usr/include/c++/6.2.1/functional:55,
                 from feature_index.cpp:23:
/usr/include/c++/6.2.1/bits/stl_pair.h:425:5: note: candidate: template<class
_T1, class _T2> constexpr std::pair<typename
std::__decay_and_strip<_Tp>::__type, typename
std::__decay_and_strip<_T2>::__type> std::make_pair(_T1&&, _T2&&)
     make_pair(_T1&& __x, _T2&& __y)
     ^~~~~~~~~
/usr/include/c++/6.2.1/bits/stl_pair.h:425:5: note:   template argument
deduction/substitution failed:
feature_index.cpp:150:51: note:   cannot convert
'rit.std::_Rb_tree_const_iterator<_Tp>::operator*<int>()' (type 'const int') to
type 'int&&'
    feature_set.insert (std::make_pair <int, int>( *rit, *cit ) );
                                                   ^~~~
feature_index.cpp:154:66: error: no matching function for call to
'make_pair(const int&, const int&)'
    bow_feature_set.insert (std::make_pair <int, int>( *rit, *cit ) );
                                                                  ^
In file included from /usr/include/c++/6.2.1/utility:70:0,
                 from /usr/include/c++/6.2.1/tuple:38,
                 from /usr/include/c++/6.2.1/functional:55,
                 from feature_index.cpp:23:
/usr/include/c++/6.2.1/bits/stl_pair.h:425:5: note: candidate: template<class
_T1, class _T2> constexpr std::pair<typename
std::__decay_and_strip<_Tp>::__type, typename
std::__decay_and_strip<_T2>::__type> std::make_pair(_T1&&, _T2&&)
     make_pair(_T1&& __x, _T2&& __y)
     ^~~~~~~~~
/usr/include/c++/6.2.1/bits/stl_pair.h:425:5: note:   template argument
deduction/substitution failed:
feature_index.cpp:154:55: note:   cannot convert
'rit.std::_Rb_tree_const_iterator<_Tp>::operator*<int>()' (type 'const int') to
type 'int&&'
    bow_feature_set.insert (std::make_pair <int, int>( *rit, *cit ) );
                                                       ^~~~
Makefile:335: ターゲット 'feature_index.lo' のレシピで失敗しました
make[2]: *** [feature_index.lo] エラー 1
make[2]: ディレクトリ '/home/xxx/yamcha-0.33/src' から出ます
Makefile:302: ターゲット 'all-recursive' のレシピで失敗しました
make[1]: *** [all-recursive] エラー 1
make[1]: ディレクトリ '/home/xxx/yamcha-0.33' から出ます
Makefile:200: ターゲット 'all' のレシピで失敗しました
make: *** [all] エラー 2
```

これは何を修正すればよいのでしょうか？

--- a/script/setup.sh
+++ b/script/setup.sh
@@ -71,7 +71,7 @@ if [ -z $NO_YAMCHA ]; then
         make install
         cd ..
         if [ ! -x ${PREFIX}/bin/yamcha ]; then
-            echo "*** APPLY Yamcha Patch for INSTALL Error ***"
+            echo "*** APPLY Yamcha Patch(1) for INSTALL Error ***"
             wget "https://gist.githubusercontent.com/skozawa/89024693963fd0adfa6d/raw/00ffa28de5ef11b902b4f35cbf3f3217bc62de3e/yamcha.patch"
             patch -p0 < yamcha.patch
             cd yamcha-0.33
@@ -80,6 +80,17 @@ if [ -z $NO_YAMCHA ]; then
             make install
             cd ..
             rm yamcha.patch
+        fi
+        if [ ! -x ${PREFIX}/bin/yamcha ]; then
+            echo "*** APPLY Yamcha Patch(2) for INSTALL Error ***"
+            cd yamcha-0.33
+            wget "https://sources.debian.org/data/main/y/yamcha/0.33-2/debian/patches/1011_fix_gcc7_compilation.patch"
+            patch -p1 < 1011_fix_gcc7_compilation.patch
+            ./configure --prefix=${PREFIX} --with-svm-learn=${PREFIX}
+            make
+            make install
+            cd ..
+            rm 1011_fix_gcc7_compilation.patch
         fi
         rm -rf yamcha-0.33
         rm -f yamcha.tar.gz

